MK404-build.sh

```diff
@@ -221,3 +312,115 @@ if [[ ! -f "$MK404_BUILD_PATH/Prusa_MK3S_SDcard.bin" || "$force_flag" == "1" ]];
 
+
+# Prepare run MK404
+#Check MK404_Printer
+MK404_PRINTER_TEMP=$(echo $firmware_version_flag | sed 's/\(.*\)\///' | grep 'MK3')
+if [ ! -z $MK404_PRINTER_TEMP ]; then
+    MK404_PRINTER=MK3
+fi
+MK404_PRINTER_TEMP=$(echo $firmware_version_flag | sed 's/\(.*\)\///' | grep 'MK3S')
+if [ ! -z $MK404_PRINTER_TEMP ]; then
+    MK404_PRINTER=MK3S
+fi
+MK404_PRINTER_TEMP=$(echo $firmware_version_flag | sed 's/\(.*\)\///' | grep 'MK25')
+if [ ! -z $MK404_PRINTER_TEMP ]; then
+    MK404_PRINTER=MK25
+fi
+MK404_PRINTER_TEMP=$(echo $firmware_version_flag | sed 's/\(.*\)\///' | grep 'MK25S')
+if [ ! -z $MK404_PRINTER_TEMP ]; then
+    MK404_PRINTER=MK25S
+fi
+if [ -z "$MK404_PRINTER" ]; then
+    echo "Tried to determine MK404 printer from hex file, but failed!"
+    echo "Add argument -p with 'MK25', 'MK25S', 'MK3' or 'MK3S' to start MK404"
+fi
+
+if [ ! -z $mk404_printer_flag ]; then
+    if [ "$mk404_printer_flag" != "$MK404_PRINTER" ]; then
+        echo "$(tput setaf 3)You defined a different printer type than the firmware!"
+        echo "This can cause unexpected issues.$(tput sgr 0)"
+        echo 
+        PS3="Select $(tput setaf 2)printer$(tput sgr 0) you want to use."
+        select which in "$(tput setaf 2)$MK404_PRINTER$(tput sgr 0)" "$mk404_printer_flag"; do
+            case $which in
+                $MK404_PRINTER)
+                    echo "Set $MK404_PRINTER as printer"
+                    break
+                    ;;
+                $mk404_printer_flag)
+                    echo "Set $(tput setaf 3)$mk404_printer_flag$(tput sgr 0) as printer"
+                    echo "$(tput setaf 3)This firmware file isn't correct for this printer!!!$(tput sgr 0)"
+                    echo 
+                    MK404_PRINTER=$mk404_printer_flag
+                    read -p "Press Enter to continue."
+                    break
+                    ;;
+                *)
+                    break
+                    ;;
+            esac
+        done
+
+    fi
+fi
+
+if [ -z $MK404_PRINTER ]; then
+    exit 7
+fi
+
+if [[ "$MK404_PRINTER" == "MK25" || "$MK404_PRINTER" == "MK25S" ]]; then
+    MK404_PRINTER="${MK404_PRINTER}_mR13"
+else
+    if [[ "$mk404_flag" == "2" || "$mk404_flag" == "MMU2" || "$mk404_flag" == "MMU2S" ]]; then # Check if MMU2 is selected only for MK3/S
+        MK404_PRINTER="${MK404_PRINTER}MMU2"
+    fi
+fi
+
+# Run MK404 with 'debugcore' and/or 'bootloader-file'
+    if [[ ! -z $MK404_DEBUG && "$MK404_DEBUG" == "atmega404" || ! -z $BOARD_MEM && "$BOARD_MEM" == "0xFFFF" ]]; then
+        MK404_options="--debugcore"
+    fi
+    if [[ ! -z $MK404_DEBUG && "$MK404_DEBUG" == "atmega404_no_bootloader"  || ! -z $BOARD_FLASH && "$BOARD_FLASH" != "0x3FFFF" ]]; then
+        MK404_options='--debugcore --bootloader-file ""'
+    fi
+
+# Run MK404 with grafics
+    if [ ! -z "$graphics_flag" ]; then
+        if [ ! -z "$MK404_options" ]; then
+            MK404_options="${MK404_options} --colour-extrusion --extrusion Quad_HR -g "
```

Comment:
Not sure if we want to make Quad_HR the default; perhaps better as an option too since the HRs can put a pretty hefty toll on the system for larger test prints.